src/openfermion/measurements/_qubit_partitioning.py

```diff
@@ -21,7 +21,7 @@
     from itertools import izip_longest as zip_longest
 
 
-def binary_partition_iterator(qubit_list, num_iterations=None):
+def binary_partition_iterator(qubit_list: list, num_iterations: int = None):
```

Comment:
This should be `List` (capital L) and will require `from typing import List`. However, this is moot if we want to continue to support python 2, since then we'd have to put all type annotations in comments.

Reply:
We've decided to drop support for Python 2.